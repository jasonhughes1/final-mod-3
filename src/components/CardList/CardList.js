import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';

const CardList = ({ allTheHouses }) => {
  const houseData = allTheHouses.map( (house, index) =>
    <Card house={house}
      key={`house - ${index}`} />);


  return (
    <div>
      <h4>{houseData}</h4>
    </div>

  );
};

CardList.propTypes = {
  allTheHouses: PropTypes.arrayOf(PropTypes.object)
};

export default CardList;
