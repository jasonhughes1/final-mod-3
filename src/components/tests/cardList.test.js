import React from 'react';
import CardList from '../CardList/CardList';
import { shallow } from 'enzyme';
import houseMock from './mockData/houseMock';

describe('CardList Tests', () => {

  it('Should render component and match snapshot', () => {
    const renderedCardList = shallow(
      <CardList allTheHouses={houseMock} />);

    expect(renderedCardList).toMatchSnapshot();
  });
});
