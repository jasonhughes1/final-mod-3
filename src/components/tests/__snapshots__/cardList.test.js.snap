// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList Tests Should render component and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    allTheHouses={
        Array [
            Object {
              "ancestralWeapons": Array [
                "Lady Forlorn",
              ],
              "coatOfArms": "Three black...",
              "diedOut": "",
              "founded": "",
              "name": "House Corbray of Heart's Home",
              "seats": Array [
                "Heart's Home",
              ],
              "titles": Array [
                "Lord of Heart's Home",
                "King of the Fingers (historical)",
              ],
            },
            Object {
              "ancestralWeapons": Array [
                "Dawn",
              ],
              "coatOfArms": "A white sword falling...",
              "diedOut": "",
              "founded": "",
              "name": "House Dayne of Starfall",
              "seats": Array [
                "Starfall",
              ],
              "titles": Array [
                "Lord of Starfall",
                "King of the Torrentine (historical)",
              ],
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h4
        className="card-container"
>
        <Card
                house={
                        Object {
                                "ancestralWeapons": Array [
                                  "Lady Forlorn",
                                ],
                                "coatOfArms": "Three black...",
                                "diedOut": "",
                                "founded": "",
                                "name": "House Corbray of Heart's Home",
                                "seats": Array [
                                  "Heart's Home",
                                ],
                                "titles": Array [
                                  "Lord of Heart's Home",
                                  "King of the Fingers (historical)",
                                ],
                              }
                }
        />
        <Card
                house={
                        Object {
                                "ancestralWeapons": Array [
                                  "Dawn",
                                ],
                                "coatOfArms": "A white sword falling...",
                                "diedOut": "",
                                "founded": "",
                                "name": "House Dayne of Starfall",
                                "seats": Array [
                                  "Starfall",
                                ],
                                "titles": Array [
                                  "Lord of Starfall",
                                  "King of the Torrentine (historical)",
                                ],
                              }
                }
        />
</h4>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            house={
                        Object {
                                    "ancestralWeapons": Array [
                                      "Lady Forlorn",
                                    ],
                                    "coatOfArms": "Three black...",
                                    "diedOut": "",
                                    "founded": "",
                                    "name": "House Corbray of Heart's Home",
                                    "seats": Array [
                                      "Heart's Home",
                                    ],
                                    "titles": Array [
                                      "Lord of Heart's Home",
                                      "King of the Fingers (historical)",
                                    ],
                                  }
            }
/>,
          <Card
            house={
                        Object {
                                    "ancestralWeapons": Array [
                                      "Dawn",
                                    ],
                                    "coatOfArms": "A white sword falling...",
                                    "diedOut": "",
                                    "founded": "",
                                    "name": "House Dayne of Starfall",
                                    "seats": Array [
                                      "Starfall",
                                    ],
                                    "titles": Array [
                                      "Lord of Starfall",
                                      "King of the Torrentine (historical)",
                                    ],
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "house - 0",
          "nodeType": "class",
          "props": Object {
            "house": Object {
              "ancestralWeapons": Array [
                "Lady Forlorn",
              ],
              "coatOfArms": "Three black...",
              "diedOut": "",
              "founded": "",
              "name": "House Corbray of Heart's Home",
              "seats": Array [
                "Heart's Home",
              ],
              "titles": Array [
                "Lord of Heart's Home",
                "King of the Fingers (historical)",
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "house - 1",
          "nodeType": "class",
          "props": Object {
            "house": Object {
              "ancestralWeapons": Array [
                "Dawn",
              ],
              "coatOfArms": "A white sword falling...",
              "diedOut": "",
              "founded": "",
              "name": "House Dayne of Starfall",
              "seats": Array [
                "Starfall",
              ],
              "titles": Array [
                "Lord of Starfall",
                "King of the Torrentine (historical)",
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "h4",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h4
          className="card-container"
>
          <Card
                    house={
                              Object {
                                        "ancestralWeapons": Array [
                                          "Lady Forlorn",
                                        ],
                                        "coatOfArms": "Three black...",
                                        "diedOut": "",
                                        "founded": "",
                                        "name": "House Corbray of Heart's Home",
                                        "seats": Array [
                                          "Heart's Home",
                                        ],
                                        "titles": Array [
                                          "Lord of Heart's Home",
                                          "King of the Fingers (historical)",
                                        ],
                                      }
                    }
          />
          <Card
                    house={
                              Object {
                                        "ancestralWeapons": Array [
                                          "Dawn",
                                        ],
                                        "coatOfArms": "A white sword falling...",
                                        "diedOut": "",
                                        "founded": "",
                                        "name": "House Dayne of Starfall",
                                        "seats": Array [
                                          "Starfall",
                                        ],
                                        "titles": Array [
                                          "Lord of Starfall",
                                          "King of the Torrentine (historical)",
                                        ],
                                      }
                    }
          />
</h4>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              house={
                            Object {
                                          "ancestralWeapons": Array [
                                            "Lady Forlorn",
                                          ],
                                          "coatOfArms": "Three black...",
                                          "diedOut": "",
                                          "founded": "",
                                          "name": "House Corbray of Heart's Home",
                                          "seats": Array [
                                            "Heart's Home",
                                          ],
                                          "titles": Array [
                                            "Lord of Heart's Home",
                                            "King of the Fingers (historical)",
                                          ],
                                        }
              }
/>,
            <Card
              house={
                            Object {
                                          "ancestralWeapons": Array [
                                            "Dawn",
                                          ],
                                          "coatOfArms": "A white sword falling...",
                                          "diedOut": "",
                                          "founded": "",
                                          "name": "House Dayne of Starfall",
                                          "seats": Array [
                                            "Starfall",
                                          ],
                                          "titles": Array [
                                            "Lord of Starfall",
                                            "King of the Torrentine (historical)",
                                          ],
                                        }
              }
/>,
          ],
          "className": "card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "house - 0",
            "nodeType": "class",
            "props": Object {
              "house": Object {
                "ancestralWeapons": Array [
                  "Lady Forlorn",
                ],
                "coatOfArms": "Three black...",
                "diedOut": "",
                "founded": "",
                "name": "House Corbray of Heart's Home",
                "seats": Array [
                  "Heart's Home",
                ],
                "titles": Array [
                  "Lord of Heart's Home",
                  "King of the Fingers (historical)",
                ],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "house - 1",
            "nodeType": "class",
            "props": Object {
              "house": Object {
                "ancestralWeapons": Array [
                  "Dawn",
                ],
                "coatOfArms": "A white sword falling...",
                "diedOut": "",
                "founded": "",
                "name": "House Dayne of Starfall",
                "seats": Array [
                  "Starfall",
                ],
                "titles": Array [
                  "Lord of Starfall",
                  "King of the Torrentine (historical)",
                ],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "h4",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
